src/chemistry/aerosol/cloud_aqueous_chemistry.F90

```diff
@@ -0,0 +1,894 @@
+       id_HO2 = get_spc_ndx( 'HO2' )
+    endif
+
+    inv_o3 = get_inv_ndx( 'O3' ) > 0
+    if (inv_o3) then
+       id_o3 = get_inv_ndx( 'O3' )
+    else
+       id_o3 = get_spc_ndx( 'O3' )
+    endif
+    inv_ho2 = get_inv_ndx( 'HO2' ) > 0
+    if (inv_ho2) then
+       id_ho2 = get_inv_ndx( 'HO2' )
+    else
+       id_ho2 = get_spc_ndx( 'HO2' )
+    endif
+
+    has_sox = (id_so2>0) .and. (id_h2o2>0) .and. (id_o3>0) .and. (id_ho2>0)
+    if (cloud_borne) then
+       has_sox = has_sox .and. (id_h2so4>0)
+    else
+       has_sox = has_sox .and. (id_so4>0) .and. (id_nh3>0)
+    endif
+
+    if (masterproc) then
+       write(iulog,*) 'sox_inti: has_sox = ',has_sox
+    endif
+
+    if( has_sox ) then
+       if (masterproc) then
+          write(iulog,*) '-----------------------------------------'
+          write(iulog,*) ' mo_setsox will do sox aerosols'
+          write(iulog,*) '-----------------------------------------'
+       endif
+    else
+       if (masterproc) then
+          write(iulog,*) '-----------------------------------------'
+          write(iulog,*) ' mo_setsox will not do sox aerosols'
+          write(iulog,*) '-----------------------------------------'
+       endif
+       return
+    end if
+
+    call sox_cldaero_init()
+
+  end subroutine sox_inti
+
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+  subroutine setsox( state, &
+       pbuf,   &
+       ncol,   &
+       lchnk,  &
+       loffset,&
+       dtime,  &
+       press,  &
+       pdel,   &
+       tfld,   &
+       mbar,   &
+       lwc,    &
+       cldfrc, &
+       cldnum, &
+       xhnm,   &
+       invariants, &
+       qcw,    &
+       qin,    &
+       xphlwc, &
+       aqso4,  &
+       aqh2so4,&
+       aqso4_h2o2, &
+       aqso4_o3,   &
+       yph_in,  &
+       aqso4_h2o2_3d, &
+       aqso4_o3_3d &
+       )
+
+    !-----------------------------------------------------------------------
+    !          ... Compute heterogeneous reactions of SOX
+    !
+    !       (0) using initial PH to calculate PH
+    !           (a) HENRYs law constants
+    !           (b) PARTIONING
+    !           (c) PH values
+    !
+    !       (1) using new PH to repeat
+    !           (a) HENRYs law constants
+    !           (b) PARTIONING
+    !           (c) REACTION rates
+    !           (d) PREDICTION
+    !-----------------------------------------------------------------------
+    !
+    use ppgrid,          only : pcols, pver
+    use chem_mods,       only : gas_pcnst, nfs
+    use chem_mods,       only : adv_mass
+    use physconst,       only : mwdry, gravit
+    use mo_constants,    only : pi
+#ifdef USE_MAM
+    use mam_clouds,      only : sox_cldaero_update, sox_cldaero_create_obj, sox_cldaero_destroy_obj
+#endif
+#ifdef USE_CLOUDS
+    use carma_clouds,    only : sox_cldaero_update, sox_cldaero_create_obj, sox_cldaero_destroy_obj
+#endif
```

Comment:
How are the CPP conditional blocks being used in conjunction with the abstract aerosol interface? It seems like we might want to make sure we're using a consistent approach to permitting the use of multiple aerosol modules.

Reply:
after some discussion, we should look into how `USE_CLOUDS` and `USE_MAM` are being set and used